ibm/service/eventnotification/resource_ibm_en_slack_template.go

```diff
@@ -0,0 +1,254 @@
+// Licensed under the Mozilla Public License v2.0
+
+package eventnotification
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	en "github.com/IBM/event-notifications-go-admin-sdk/eventnotificationsv1"
+	"github.com/IBM/go-sdk-core/v5/core"
+)
+
+func ResourceIBMEnSlackTemplate() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceIBMEnSlackTemplateCreate,
+		ReadContext:   resourceIBMEnSlackTemplateRead,
+		UpdateContext: resourceIBMEnSlackTemplateUpdate,
+		DeleteContext: resourceIBMEnSlackTemplateDelete,
+		Importer:      &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"instance_guid": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique identifier for IBM Cloud Event Notifications instance.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Subscription name.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Subscription description.",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The type of template",
+			},
+			"params": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"body": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The Slack Template body.",
+						},
+					},
+				},
+			},
+			"template_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Subscription ID.",
+			},
+			"subscription_count": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "Number of subscriptions.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Last updated time.",
+			},
+			"subscription_names": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of subscriptions.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceIBMEnSlackTemplateCreate(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	enClient, err := meta.(conns.ClientSession).EventNotificationsApiV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	options := &en.CreateTemplateOptions{}
+
+	options.SetInstanceID(d.Get("instance_guid").(string))
+
+	options.SetName(d.Get("name").(string))
+	options.SetType(d.Get("type").(string))
+
+	if _, ok := d.GetOk("description"); ok {
+		options.SetDescription(d.Get("description").(string))
+	}
+
+	params := EmailTemplateParamsMap(d.Get("params.0").(map[string]interface{}))
```

Comment:
params is optional accessing directly with index zero crashes plugin when params are n't provided in tempaltes